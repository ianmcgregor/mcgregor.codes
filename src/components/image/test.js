import React from 'react';
import {render} from 'enzyme';
import Image from './';

const props = {
    image: {
        id: 'c3Bh6RazkvSucssagW2GquS',
        title: 'comic-adventure',
        description: '',
        resolutions: {
            aspectRatio: 1.7777777777777777,
            width: 400,
            height: 225,
            src: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&q=50',
            srcSet: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&h=225&q=50 1x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=600&h=338&q=50 1.5x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&h=450&q=50 2x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1200&h=675&q=50 3x',
            srcWebp: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&q=50&fm=webp',
            srcSetWebp: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&h=225&q=50&fm=webp 1x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=600&h=338&q=50&fm=webp 1.5x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&h=450&q=50&fm=webp 2x,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1200&h=675&q=50&fm=webp 3x'
        },
        file: {
            url: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png',
            fileName: 'comic-adventure.png',
            contentType: 'image/png'
        },
        sizes: {
            aspectRatio: 1.7777777777777777,
            src: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&q=100',
            srcSet: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=200&h=113&q=100 200w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&h=225&q=100 400w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&h=450&q=100 800w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1200&h=675&q=100 1200w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1600&h=900&q=100 1600w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2400&h=1350&q=100 2400w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2560&h=1440&q=100 2560w',
            srcWebp: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&q=100&fm=webp',
            srcSetWebp: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=200&h=113&q=100&fm=webp 200w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&h=225&q=100&fm=webp 400w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&h=450&q=100&fm=webp 800w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1200&h=675&q=100&fm=webp 1200w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1600&h=900&q=100&fm=webp 1600w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2400&h=1350&q=100&fm=webp 2400w,\n//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2560&h=1440&q=100&fm=webp 2560w',
            sizes: '(max-width: 800px) 100vw, 800px'
        },
        resize: {
            src: '//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&fl=progressive&q=50',
            width: 400,
            height: 225,
            aspectRatio: 1.7777777777777777
        }
    }
};

describe('Image component', () => {
    it('should render', () => {
        const el = render((
            <Image {...props}/>
        ));
        expect(el).toMatchSnapshot();
    });
});
