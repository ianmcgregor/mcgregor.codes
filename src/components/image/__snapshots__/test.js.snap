// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa cpqOBv",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH gVOSqX",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "comic-adventure",
              "class": "sc-htpNat dZHANp",
              "src": "//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&q=100",
              "srcset": "//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=200&h=113&q=100 200w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=400&h=225&q=100 400w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=800&h=450&q=100 800w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1200&h=675&q=100 1200w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=1600&h=900&q=100 1600w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2400&h=1350&q=100 2400w,
//images.ctfassets.net/i8x1lkvuio7i/3Bh6RazkvSucssagW2GquS/a260280a7ec428b7a7ed1dec713df2a1/comic-adventure.png?w=2560&h=1440&q=100 2560w",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
              "srcset": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
              "srcset": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
